modules/nlu/src/backend/legacy-election.ts

```diff
@@ -0,0 +1,169 @@
+import * as sdk from 'botpress/sdk'
+import * as math from './tools/math'
+import _ from 'lodash'
+
+const OOS_AS_NONE_TRESH = 0.3
+const LOW_INTENT_CONFIDENCE_TRESH = 0.4
+const NONE_INTENT = 'none' // should extract constant in comon place
+
+type EventUnderstanding = sdk.IO.EventUnderstanding
+
+export default async function elect(predictionOutput: EventUnderstanding): Promise<EventUnderstanding> {
```

Comment:
To me this whole process can be sync.

Can you add

const debug = DEBUG('nlu.prediciton' )
and debug(nlu)  right after election.

I don't remember the exact syntax, check for debugging statements in training pipeline